KS-2-5

KinCadeFoster opened the pull request:
- расчет сложных процентов

KinCadeFoster committed:
KS-2-5
- расчет сложных процентов

ievetrovdev commented on src/main/kotlin/lesson_2/lesson2_task5.kt:
```diff
@@ -0,0 +1,11 @@
+
+const val DEPOSIT_AMOUNT = 70000
+const val INTEREST_RATE = 16.7
+const val DEPOSIT_TERM_YEARS = 20
```
Опять константы

ievetrovdev commented on src/main/kotlin/lesson_2/lesson2_task5.kt:
```diff
@@ -0,0 +1,11 @@
+const val DEPOSIT_TERM_YEARS = 20
+
+fun main(){
+    val amount = DEPOSIT_AMOUNT.toDouble() * Math.pow(1 + INTEREST_RATE / 100, DEPOSIT_TERM_YEARS.toDouble())
```
Переводить в дробные не обязательно, можно использовать функцию расширения pow на число 1 + INTEREST_RATE / 100

KinCadeFoster committed:
KS-2-5
- переработал (1 + interestRate / 100).pow(depositTermYears.toDouble())
- убрал константы

ievetrovdev commented on src/main/kotlin/lesson_2/lesson2_task5.kt:
```diff
@@ -0,0 +1,13 @@
+    val depositAmount = 70000
+    val interestRate = 16.7
+    val depositTermYears = 20
+    val amount = depositAmount.toDouble() * (1 + interestRate / 100).pow(depositTermYears.toDouble())
```
Почти верно, перевод depositAmount и depositTermYears в дробные лишние операции. 
так же можно отделить пустой строкой от блока где просто объявлены значения

KinCadeFoster committed:
KS-2-5
- привел оформленеие кода в порядок
- убрал перевод в дробные части при расчете amount
